tools/generator/Tests/Unit-Tests/EnumGeneratorTests.cs

```diff
@@ -0,0 +1,96 @@
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Reflection;
+using System.Text;
+using MonoDroid.Generation;
+using NUnit.Framework;
+using NUnit.Framework.Internal;
+
+namespace generatortests.Unit_Tests
+{
+	[TestFixture]
+	class EnumGeneratorTests
+	{
+		protected EnumGenerator generator;
+		protected StringBuilder builder;
+		protected StringWriter writer;
+
+		[SetUp]
+		public void SetUp ()
+		{
+			builder = new StringBuilder ();
+			writer = new StringWriter (builder);
+
+			generator = new EnumGenerator (writer);
+		}
+
+		[Test]
+		public void WriteBasicEnum ()
+		{
+			var enu = CreateEnum ();
+			enu.Value.FieldsRemoved = true;
+
+			generator.WriteEnumeration (enu, null);
+
+			Assert.AreEqual (GetExpected (nameof (WriteBasicEnum)), writer.ToString ().NormalizeLineEndings ());
+		}
+
+		[Test]
+		public void WriteFlagsEnum ()
+		{
+			var enu = CreateEnum ();
+			enu.Value.BitField = true;
+			enu.Value.FieldsRemoved = true;
+
+			generator.WriteEnumeration (enu, null);
+
+			Assert.AreEqual (GetExpected (nameof (WriteFlagsEnum)), writer.ToString ().NormalizeLineEndings ());
+		}
+
+		[Test]
+		public void WriteEnumWithGens ()
+		{
+			var enu = CreateEnum ();
+			var gens = CreateGens ();
+
+			generator.WriteEnumeration (enu, gens);
+			
+			Assert.AreEqual (GetExpected (nameof (WriteEnumWithGens)), writer.ToString ().NormalizeLineEndings ());
+		}
+
+		protected string GetExpected (string testName)
+		{
+			var root = Path.GetDirectoryName (Assembly.GetExecutingAssembly ().Location);
+
+			return File.ReadAllText (Path.Combine (root, "Unit-Tests", "EnumGeneratorExpectedResults", $"{testName}.txt")).NormalizeLineEndings ();
+		}
+
+		KeyValuePair<string, EnumMappings.EnumDescription> CreateEnum ()
+		{
+			var enu = new EnumMappings.EnumDescription {
+				Members = new Dictionary<string, string> {
```

Comment:
Entirely unrelated, this `Members` and `JniNames` APIs are ugly; they need to be consistent with each other, but they're entirely separate properties, and thus need not *actually* be consistent. :-(

(This doesn't need to be fixed here; it's simply that I noticed this now.)